manageability-and-operations/observability-and-manageability/oci-monitoring/custom-metrics/custom-metric-python-SDK-services-limit-monitoring/files/Scripts/postServiceLimitsMetricsIAM.py

```diff
@@ -0,0 +1,300 @@
+# use, sell, offer for sale, import, export, have made, and have sold the
+# Software and the Larger Work(s), and to sublicense the foregoing rights on
+# either these or other terms.
+
+# This license is subject to the following condition:
+# The above copyright notice and either this complete permission notice or at
+# a minimum a reference to the UPL must be included in all copies or
+# substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+###
+# This is a sample python script that post a custom metric (service_limits) to oci monitoring based on Tenancy Service Limits information.
+# Run this script on any host with python with access to your tenancy.
+# Command: python3 postServiceLimitsMetricsIAM.py
+# Version: 0.1
+###
+
+import oci,datetime,json
+from pytz import timezone
+
+# Using IAM user credentials and default profile (~/.oci/config)
+from oci.config import from_file
+config = from_file()
+
+# Vars:
+# Replace here with your tenancy's root compartment OCID
+compartment_ocid = "ocid1.tenancy.oc1....."
+
+# Start:
+now = datetime.datetime.now().strftime("%d/%m/%Y %H:%M:%S")
+print("[", now,"] Starting OCI Service Metrics limits gathering and customer metrics post...")
+
+# We gather the list of availability domains in the region
+identity_client = oci.identity.IdentityClient(config)
+list_availability_domains_response = identity_client.list_availability_domains(compartment_id = compartment_ocid)
+
+# Get the data from response
+print(list_availability_domains_response.data)
+
+# Initialize service client with default config file
+monitoring_client = oci.monitoring.MonitoringClient(config,service_endpoint="https://telemetry-ingestion.eu-frankfurt-1.oraclecloud.com")
+
+# Get Service Limits
+
+# Initialize service client with default config file
+limits_client = oci.limits.LimitsClient(config)
+
+# Send the request to service, some parameters are not required, see API
+# doc for more info
+list_limit_definitions_response = limits_client.list_limit_definitions(
+    compartment_id = compartment_ocid,
+    sort_by="name",
+    sort_order="ASC")
+
+# We iterate the list of all the service limits
+for x in list_limit_definitions_response.data:
+    limit = json.loads(str(x))
+    s_name = limit["service_name"]
+    l_name = limit["name"]
+    l_scope = limit["scope_type"]
+
+    print ("Service: ", s_name, "Limit: ", l_name, "Scope: ", l_scope)
+    
+    # If the resource limit has an AD scope, we have to specify the AD or we'll get an API 400 response
+    if l_scope == "AD" :
```

Comment:
if and else have some parts in common, too. It might be worth to factor these in a function.